services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -442,8 +443,10 @@ private void validateTokenReuseForRefresh(KeycloakSession session, RealmModel re
             AuthenticatedClientSessionModel clientSession = validation.clientSessionCtx.getClientSession();
             try {
                 validateTokenReuse(session, realm, refreshToken, clientSession, true);
-                int currentCount = clientSession.getCurrentRefreshTokenUseCount();
-                clientSession.setCurrentRefreshTokenUseCount(currentCount + 1);
+                String key = Optional.ofNullable(refreshToken.getOtherClaims().get(Constants.REUSE_ID)).map(String::valueOf).orElse("");
+                int currentCount = clientSession.getRefreshTokenUseCount(key);
+                clientSession.setRefreshTokenUseCount(key, currentCount + 1);
+                clientSession.setRefreshTokenLastRefresh(key, Time.currentTime());
```

Comment:
I can think of the corner case that `Time.currentTime()`, which is generated on this line maybe be slightly different from the time in the `iat` claim of the refresh token. It seems there can be some small difference (like EG. 1 second) among them. I can imagine this can cause some issues in the corner scenarios...

I've noticed that similar issue already exists on this line https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java#L1127 . Is it possible to update this to use something like `int currentTime = refreshToken.getISsuedAt();` to make sure that it is really same time in the refreshToken and clientSession/userSession and also make sure that `clientSession.setRefreshTokenLastRefresh` is updated with this timestamp as well?

Maybe this is a nitpick, but from the past experience, the similar things can cause troubles in production or security issues or flaky tests in our CI (and maybe this one will fix some flaky tests as a side-effect :-)